Refactor plugin

omrisarig13 opened the pull request:
I refactored the whole plugin, to make it more suitable to today plugin's standard in vim.

I didn't change almost any functionality of the plugin, all the functionality I changed was fixing TODO messages in the code and small fixes of bugs I found on the way.

omrisarig13 committed:
Refactored the plugin code.

Change the directory structure to be as the needed directory structure
of a vim plugin, and refactor the code.
When refactoring the code, I mainly moved the functions to autoload
directory, and set the namespace for them. I did little changes to the
functions themselves, and most of them were TODO fixes.

I didn't add or remove any functionality from the code itself.

omrisarig13 committed:
Update readme.md file

Changed the format of the readme file to be github's markdown format, instead of a plain text file.

omrisarig13 committed:
Added doc directory.

Added a documents directory. This directory contains the needed files
for vim to be able to show help on the marvim functions and commands as
well.

omrisarig13 committed:
Typo fixes and redundant docs removed.

Moved over the code and fixed all the typos that left in it.
Also removed the documentation from the plugin/marvim.vim file, since it
all is in the readme file.

chamindra commented:
Hi Omri,

Sorry I could not reply earlier. I had a busier weekend than planned. I
have approved the merge request now. How much have you tested it so far?

Chamindra de Silva
https://www.linkedin.com/in/chamindra |
http://chamindra-de-silva.blogspot.com


On Sat, Jun 15, 2019 at 2:23 PM Omri Sarig <notifications@github.com> wrote:

> I refactored the whole plugin, to make it more suitable to today plugin's
> standard in vim.
>
> I didn't change almost any functionality of the plugin, all the
> functionality I changed was fixing TODO messages in the code and small
> fixes of bugs I found on the way.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/chamindra/marvim/pull/3
> Commit Summary
>
>    - Refactored the plugin code.
>    - Update readme.md file
>    - Added doc directory.
>    - Typo fixes and redundant docs removed.
>
> File Changes
>
>    - *D* README <https://github.com/chamindra/marvim/pull/3/files#diff-0>
>    (166)
>    - *A* autoload/marvim.vim
>    <https://github.com/chamindra/marvim/pull/3/files#diff-1> (333)
>    - *A* doc/marvim.txt
>    <https://github.com/chamindra/marvim/pull/3/files#diff-2> (180)
>    - *A* doc/tags
>    <https://github.com/chamindra/marvim/pull/3/files#diff-3> (27)
>    - *D* marvim.vim
>    <https://github.com/chamindra/marvim/pull/3/files#diff-4> (440)
>    - *A* plugin/marvim.vim
>    <https://github.com/chamindra/marvim/pull/3/files#diff-5> (67)
>    - *A* readme.md
>    <https://github.com/chamindra/marvim/pull/3/files#diff-6> (164)
>
> Patch Links:
>
>    - https://github.com/chamindra/marvim/pull/3.patch
>    - https://github.com/chamindra/marvim/pull/3.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/chamindra/marvim/pull/3?email_source=notifications&email_token=ABTG6G443XHN4KX3ZQ63VI3P2TUNRA5CNFSM4HYPA7U2YY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4GZWGVHQ>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABTG6GZEI4DALFLDNQSUCQ3P2TUNRANCNFSM4HYPA7UQ>
> .
>

chamindra commented:
Thanks Omri,

I am going to publish this as v0.5. Can you provide me with the details for
the change log.

Chamindra de Silva
https://www.linkedin.com/in/chamindra |
http://chamindra-de-silva.blogspot.com


On Tue, Jun 18, 2019 at 3:50 AM Chamindra de Silva <chamindra@gmail.com>
wrote:

> Hi Omri,
>
> Sorry I could not reply earlier. I had a busier weekend than planned. I
> have approved the merge request now. How much have you tested it so far?
>
> Chamindra de Silva
> https://www.linkedin.com/in/chamindra |
> http://chamindra-de-silva.blogspot.com
>
>
> On Sat, Jun 15, 2019 at 2:23 PM Omri Sarig <notifications@github.com>
> wrote:
>
>> I refactored the whole plugin, to make it more suitable to today plugin's
>> standard in vim.
>>
>> I didn't change almost any functionality of the plugin, all the
>> functionality I changed was fixing TODO messages in the code and small
>> fixes of bugs I found on the way.
>> ------------------------------
>> You can view, comment on, or merge this pull request online at:
>>
>>   https://github.com/chamindra/marvim/pull/3
>> Commit Summary
>>
>>    - Refactored the plugin code.
>>    - Update readme.md file
>>    - Added doc directory.
>>    - Typo fixes and redundant docs removed.
>>
>> File Changes
>>
>>    - *D* README <https://github.com/chamindra/marvim/pull/3/files#diff-0>
>>    (166)
>>    - *A* autoload/marvim.vim
>>    <https://github.com/chamindra/marvim/pull/3/files#diff-1> (333)
>>    - *A* doc/marvim.txt
>>    <https://github.com/chamindra/marvim/pull/3/files#diff-2> (180)
>>    - *A* doc/tags
>>    <https://github.com/chamindra/marvim/pull/3/files#diff-3> (27)
>>    - *D* marvim.vim
>>    <https://github.com/chamindra/marvim/pull/3/files#diff-4> (440)
>>    - *A* plugin/marvim.vim
>>    <https://github.com/chamindra/marvim/pull/3/files#diff-5> (67)
>>    - *A* readme.md
>>    <https://github.com/chamindra/marvim/pull/3/files#diff-6> (164)
>>
>> Patch Links:
>>
>>    - https://github.com/chamindra/marvim/pull/3.patch
>>    - https://github.com/chamindra/marvim/pull/3.diff
>>
>> —
>> You are receiving this because you are subscribed to this thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/chamindra/marvim/pull/3?email_source=notifications&email_token=ABTG6G443XHN4KX3ZQ63VI3P2TUNRA5CNFSM4HYPA7U2YY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4GZWGVHQ>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/ABTG6GZEI4DALFLDNQSUCQ3P2TUNRANCNFSM4HYPA7UQ>
>> .
>>
>

omrisarig13 commented:
Hey,

It's totally fine that you didn't answer earlier, there is no rush in
working on this.

To answer you first question, I have tested the plugin only on my machine,
but I tried to test all the features of it. I saved new macros and
templates, loaded them, and did it for multiple directories and different
file types. It seems to work.
My machine as ManjaroLinux 18.0.4 (a distribution based on Arch), with vim
8.1.1467.
I could not test the plugin on any other machine, since I have none.

About the change log. The changes I did are documented in the documentation
of the plugin (I forgot to add them to the readme as well).
The documentation are here - https://github.com/chamindra/marvim
/blob/2a28500d5fcec1710849e0536f541f1316bef963/doc/marvim.txt#L144

However, when reading and refactoring the plugin I came across some more
features/small bugs that I would want to fix.
This is the list with all the things that I wanted to do:

   - Add a specific option for calling up templates or macros separately.
   - Before running over already exists template/macro, ask the user for
   conformation.
   - If there are template file and macro file with the same name, the
   macro file will always be used. In that case, ask the user to choose its
   wanted action.
   - Add better explanation about what are templates.
   - Check vim version number when running macro
   - Add flags about whether the plugin should work for macros only,
   templates only, or both.
   - Add interface to remove macro files.
   - Add interface to edit macro files.
   - Move the example directory to another repository that would have all
   the shared macros.

I intended to open all those as issues in the github, and work them one by
one in the near future.
If you have some input on those issues I would be happy to hear it.


Omri


On Tue, Jun 18, 2019 at 12:53 PM Chamindra de Silva <
notifications@github.com> wrote:

> Thanks Omri,
>
> I am going to publish this as v0.5. Can you provide me with the details for
> the change log.
>
> Chamindra de Silva
> https://www.linkedin.com/in/chamindra |
> http://chamindra-de-silva.blogspot.com
>
>
> On Tue, Jun 18, 2019 at 3:50 AM Chamindra de Silva <chamindra@gmail.com>
> wrote:
>
> > Hi Omri,
> >
> > Sorry I could not reply earlier. I had a busier weekend than planned. I
> > have approved the merge request now. How much have you tested it so far?
> >
> > Chamindra de Silva
> > https://www.linkedin.com/in/chamindra |
> > http://chamindra-de-silva.blogspot.com
> >
> >
> > On Sat, Jun 15, 2019 at 2:23 PM Omri Sarig <notifications@github.com>
> > wrote:
> >
> >> I refactored the whole plugin, to make it more suitable to today
> plugin's
> >> standard in vim.
> >>
> >> I didn't change almost any functionality of the plugin, all the
> >> functionality I changed was fixing TODO messages in the code and small
> >> fixes of bugs I found on the way.
> >> ------------------------------
> >> You can view, comment on, or merge this pull request online at:
> >>
> >> https://github.com/chamindra/marvim/pull/3
> >> Commit Summary
> >>
> >> - Refactored the plugin code.
> >> - Update readme.md file
> >> - Added doc directory.
> >> - Typo fixes and redundant docs removed.
> >>
> >> File Changes
> >>
> >> - *D* README <https://github.com/chamindra/marvim/pull/3/files#diff-0>
> >> (166)
> >> - *A* autoload/marvim.vim
> >> <https://github.com/chamindra/marvim/pull/3/files#diff-1> (333)
> >> - *A* doc/marvim.txt
> >> <https://github.com/chamindra/marvim/pull/3/files#diff-2> (180)
> >> - *A* doc/tags
> >> <https://github.com/chamindra/marvim/pull/3/files#diff-3> (27)
> >> - *D* marvim.vim
> >> <https://github.com/chamindra/marvim/pull/3/files#diff-4> (440)
> >> - *A* plugin/marvim.vim
> >> <https://github.com/chamindra/marvim/pull/3/files#diff-5> (67)
> >> - *A* readme.md
> >> <https://github.com/chamindra/marvim/pull/3/files#diff-6> (164)
> >>
> >> Patch Links:
> >>
> >> - https://github.com/chamindra/marvim/pull/3.patch
> >> - https://github.com/chamindra/marvim/pull/3.diff
> >>
> >> —
> >> You are receiving this because you are subscribed to this thread.
> >> Reply to this email directly, view it on GitHub
> >> <
> https://github.com/chamindra/marvim/pull/3?email_source=notifications&email_token=ABTG6G443XHN4KX3ZQ63VI3P2TUNRA5CNFSM4HYPA7U2YY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4GZWGVHQ
> >,
> >> or mute the thread
> >> <
> https://github.com/notifications/unsubscribe-auth/ABTG6GZEI4DALFLDNQSUCQ3P2TUNRANCNFSM4HYPA7UQ
> >
> >> .
> >>
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/chamindra/marvim/pull/3?email_source=notifications&email_token=AKDEWM22XMTPJICXWIPRUD3P3CWCZA5CNFSM4HYPA7U2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODX53FIQ#issuecomment-503034530>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKDEWM3XYOU4CKEYV424EYTP3CWCZANCNFSM4HYPA7UQ>
> .
>
